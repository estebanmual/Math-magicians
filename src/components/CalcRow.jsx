import React from 'react';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';
import styles from './Calculator.module.css';

function CalcRow(props) {
  const { row, clickHandler } = props;

  return (
    <tr>
      {row.map((key, index) => {
        let name = styles.calculator__button__grey;
        if ((index + 1) % 4 === 0 || key === '=') { name = styles.calculator__button__orange; }
        if (key === '0') { return <td key={uuidv4()} colSpan="2" className={name}><button type="button" onClick={clickHandler}>0</button></td>; }
        return (
          <td key={uuidv4()} className={name}><button type="button" onClick={clickHandler}>{key}</button></td>
        );
      })}
    </tr>
  );
}

export default CalcRow;

CalcRow.defaultProps = {
  row: [],
};

CalcRow.propTypes = {
  row: PropTypes.arrayOf(PropTypes.string),
  clickHandler: PropTypes.func.isRequired,
};
