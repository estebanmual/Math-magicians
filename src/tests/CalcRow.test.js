import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CalcRow from '../components/CalcRow';

const mockhandler = jest.fn();
const row = ['AC', '+/-', '%', 'รท'];

it('renders correctly when there are no items', () => {
  const tree = renderer
    .create(<table><tbody><CalcRow row={row} clickHandler={mockhandler} /></tbody></table>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const button = render(<table><tbody><CalcRow row={row} clickHandler={mockhandler} /></tbody></table>).getByText('รท');
  fireEvent.click(button);

  expect(mockhandler).toHaveBeenCalledTimes(1);
});
