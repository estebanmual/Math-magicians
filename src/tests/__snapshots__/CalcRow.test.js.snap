// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<table>
  <tbody>
    <tr>
      <td
        className="calculator__button__grey"
      >
        <button
          onClick={[MockFunction]}
          type="button"
        >
          AC
        </button>
      </td>
      <td
        className="calculator__button__grey"
      >
        <button
          onClick={[MockFunction]}
          type="button"
        >
          +/-
        </button>
      </td>
      <td
        className="calculator__button__grey"
      >
        <button
          onClick={[MockFunction]}
          type="button"
        >
          %
        </button>
      </td>
      <td
        className="calculator__button__orange"
      >
        <button
          onClick={[MockFunction]}
          type="button"
        >
          รท
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
